prow/cmd/gerrit/main.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package main is a gerrit fetcher updating against a target gerrit instance
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit"
+	"k8s.io/test-infra/prow/kube"
+)
+
+var (
+	configPath = flag.String("config-path", "/etc/config/config", "Path to config.yaml.")
+
+	instance  = flag.String("gerrit-instance", "", "URL to gerrit instance")
+	projects  = flag.String("gerrit-projects", "", "Comma separated gerrit projects to fetch from the gerrit instance")
+	rateLimit = flag.Int("rate-limit", 5, "How many gerrit changes to fetch per API call")
+	interval  = flag.Duration("sync-interval", 30*time.Second, "How often to sync gerrit changes")
+)
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	logger := logrus.WithField("component", "gerrit")
+
+	if *instance == "" {
+		logger.Fatal("--gerrit-intance must set")
+	}
+
+	projs := strings.Split(*projects, ",")
+	if len(projs) == 0 {
+		logger.Fatal("must have one or more target gerrit project")
+	}
+
+	// authenticate to gerrit server
+	if err := exec.Command("git", "clone", "https://").Run(); err != nil {
+		logger.Fatal("Fail to git clone https://gerrit.googlesource.com/gcompute-tools : %v", err)
+	} else if err := exec.Command("./gcompute-tools/git-cookie-authdaemon").Run(); err != nil {
+		logger.Fatal("Fail to authenticate using git-cookie-authdaemon : %v", err)
+	}
+
+	ca := &config.Agent{}
+	if err := ca.Start(*configPath); err != nil {
+		logger.WithError(err).Fatal("Error starting config agent.")
+	}
+
+	kc, err := kube.NewClientInCluster(ca.Config().ProwJobNamespace)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting kube client.")
+	}
+
+	c, err := gerrit.NewController(*instance, projs, kc, logger, ca)
+
+	tick := time.Tick(*interval)
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+
+	for {
+		select {
+		case <-tick:
+			start := time.Now()
+			if err := c.Sync(*rateLimit); err != nil {
```

Comment:
nit: Like the sync period, the rate limit could be moved to the config if tweaking the ratelimit is something we will do frequently.